dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/helpers/calculate-initial-utilisation-and-fixed-fee.test.ts

```diff
@@ -0,0 +1,218 @@
+import { calculateInitialUtilisation } from '@ukef/dtfs2-common';
+import { calculateInitialUtilisationAndFixedFee, parseDate, hasRequiredValues } from './calculate-initial-utilisation-and-fixed-fee';
+import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
+import { aTfmFacility } from '../../../../../../test-helpers';
+import { calculateInitialFixedFee } from './calculate-initial-fixed-fee';
+
+describe('helpers/calculate-initial-utilisation-and-fixed-fee', () => {
+  describe('parseDate', () => {
+    describe('date passed as null', () => {
+      it('should return an error', () => {
+        expect(() => parseDate(null)).toThrow('Invalid date');
+      });
+    });
+
+    describe('date passed in a correct format', () => {
+      it('should return the formatted date', () => {
+        const date = new Date();
+
+        const result = parseDate(date);
+
+        expect(result).toEqual(date);
+      });
+    });
+
+    describe('date passed as a string', () => {
+      it('should return the formatted date', () => {
+        const date = new Date().toString();
+
+        const result = parseDate(date);
+        const expected = new Date(date);
+
+        expect(result).toEqual(expected);
+      });
+    });
+
+    describe('date passed in EPOCH', () => {
+      it('should return the formatted date', () => {
+        const date = new Date().getTime();
+
+        const result = parseDate(date);
+        const expected = new Date(date);
+
+        expect(result).toEqual(expected);
+      });
+    });
+  });
+
+  describe('hasRequiredValues', () => {
+    it('should return true if all values are present', () => {
+      const params = {
+        value: 1,
+        interestPercentage: 2,
+        dayCountBasis: 3,
+        coverStartDate: new Date(),
+        coverEndDate: new Date(),
+      };
+
+      const result = hasRequiredValues(params);
+      expect(result).toEqual(true);
+    });
+
+    it('should return false if the value is not provided', () => {
+      const params = {
+        value: null,
+        interestPercentage: 2,
+        dayCountBasis: 3,
+        coverStartDate: new Date(),
+        coverEndDate: new Date(),
+      };
+
+      const result = hasRequiredValues(params);
+      expect(result).toEqual(false);
+    });
+
+    it('should return false if the interestPercentage is not provided', () => {
+      const params = {
+        value: 1,
+        interestPercentage: null,
+        dayCountBasis: 3,
+        coverStartDate: new Date(),
+        coverEndDate: new Date(),
+      };
+
+      const result = hasRequiredValues(params);
+      expect(result).toEqual(false);
+    });
+
+    it('should return false if the dayCountBasis is not provided', () => {
+      const params = {
+        value: 1,
+        interestPercentage: 2,
+        dayCountBasis: null,
+        coverStartDate: new Date(),
+        coverEndDate: new Date(),
+      };
```

Comment:
Could we create a `baseParams` const or something so that each test is shorter/one line params? E.g for this test:

```ts
 const params = { ...validParams, dayCountBasis: null };
```

Reply:
Could even just do this:

```ts
const result = hasRequiredValues({ ...params, dayCountBasis: null });
```

Reply:
done

Reply:
Resolving in place of a new comment